src/pure/inmost.ml

```diff
@@ -702,3 +702,35 @@ let set_cookie
   in
 
   add_header "Set-Cookie" set_cookie response
+
+let drop_cookie
+    ?prefix:cookie_prefix
+    ?domain
+    ?path
+    ?secure
+    ?(http_only = true)
+    ?same_site
+    name
+    request
+    response =
+  let set_cookie = set_cookie ~http_only:http_only ~expires:0. ~encrypt:false in
+
+  let set_cookie = match cookie_prefix with
+    | Some p -> set_cookie ~prefix:p
+    | None -> set_cookie ~prefix:None in
+  
+  let set_cookie = match path with
+    | Some p -> set_cookie ~path:p
+    | None -> set_cookie ~path:(Some (prefix request)) in
```

Comment:
Most of the arguments can be simply forwarded to `set_cookie`, e.g.

```
set_cookie (* ... *) ?path(* ... *)
```

The `?secure` syntax forwards an optional argument to an optional argument, so you don't have to pattern match on them and then build up a `set_cookie` closure gradually, as here.

I'll take care of tweaking this after merge.

Reply:
Wow thanks, I was looking for a feature like this but I could not find it anywhere.